spanner/client.go

```diff
@@ -555,6 +555,10 @@ func (c *Client) rwTransaction(ctx context.Context, f func(context.Context, *Rea
 			err error
 		)
 		if sh == nil || sh.getID() == "" || sh.getClient() == nil {
+			//  retry with explicit begin transaction
+			if t != nil {
```

Comment:
Can we have (a) test(s) that show:
1. What would have gone wrong before without this change (i.e. a test that fails without this change).
2. What is the behavior for the scenario that this fix solves (i.e. a test that succeeds even if a `Session not found` error occurs)

It might very well be possible that the two can be combined into one test.